### PR TITLE
Vertx server for artipie

### PR DESCRIPTION
**Pull request by g4s8**
#12 - implemented main class to start Artipie in Vertx server from https://github.com/artipie/vertx-server

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 PLease assign me

**Comment by g4s8** [comment]
@0crat assign @paulodamaso

**Comment by 0crat** [comment]
> @0crat assign @paulodamaso  ([here](https://github.com/artipie/artipie/issues/30#issuecomment-587454011))

@g4s8 This pull request [#30](https://github.com/artipie/artipie/issues/30) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/52380cfb-e5c5-41e8-b9ec-c1540d7c6fbb); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: src/main/java/com/artipie/VertxMain.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public void run() {
+        new VertxSliceServer(this.slice, this.port).start();
+    }
+
+    /**
+     * Entry point.
+     * @param args CLI args
+     */
+    public static void main(final String... args) {
+        final String storage = System.getProperty("artipie.storage");
```
Comment (Sammers21) [inline src/main/java/com/artipie/VertxMain.java]: Can we avoid properties and use `args`?

File: src/main/java/com/artipie/VertxMain.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                new Pie(new FileStorage(Path.of(storage))),
+                port
+            )
+        );
+        thread.setName(VertxMain.class.getSimpleName());
+        thread.checkAccess();
+        thread.setPriority(Thread.MAX_PRIORITY);
+        thread.start();
+        try {
+            thread.join();
```
Comment (Sammers21) [inline src/main/java/com/artipie/VertxMain.java]: No, don't do this. Just start `VertxSliceServer` and that is it.

File: src/main/java/com/artipie/VertxMain.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import com.artipie.http.Slice;
+import com.artipie.vertx.VertxSliceServer;
+import com.jcabi.log.Logger;
+import java.nio.file.Path;
+
+/**
+ * Vertx server entry point.
+ * @since 1.0
+ */
+public final class VertxMain implements Runnable {
```
Comment (Sammers21) [inline src/main/java/com/artipie/VertxMain.java]: Can you just call it `Main`?

**Comment by g4s8** [comment]
@0crat refuse

File: src/main/java/com/artipie/VertxMain.java (hunk 1 above)
Comment (g4s8) [reply src/main/java/com/artipie/VertxMain.java]: @Sammers21 you are right, added todo for CLI options

File: src/main/java/com/artipie/VertxMain.java (hunk 2 above)
Comment (g4s8) [reply src/main/java/com/artipie/VertxMain.java]: @Sammers21 fixed

File: src/main/java/com/artipie/VertxMain.java (hunk 3 above)
Comment (g4s8) [reply src/main/java/com/artipie/VertxMain.java]: @Sammers21 no, it's one of possible implementation, I'd prefer to use implementation name as part of class name

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/artipie/issues/30#issuecomment-589530772))

@g4s8 The user @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) resigned from [#30](https://github.com/artipie/artipie/issues/30), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#30` is not assigned, can't get performer

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**